File: 0162-find-peak-element/0162-find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int beg=0;
        int end=nums.size()-1;
        while(beg<end){
            int mid=beg+(end-beg)/2;
            if(nums[mid]>nums[mid+1]){
                end=mid;
            }else{
                
                beg=mid+1;
            }
            
        }
        return beg;
    }
};

File: 0162-find-peak-element/0162-find-peak-element.py
class Solution(object):
    def findPeakElement(self, nums):
        n=len(nums)
        for i in range(n):
            #check left side
            if i==0:
                left=True
            else:
                left=nums[i]>nums[i-1]
            #check right side
            if i==n-1:
                right=True
            else:
                right=nums[i]>nums[i+1]
            if left and right:
                return i
